common/static/common/js/discussion/utils.js

```diff
@@ -353,31 +373,38 @@
             return this.getWmdEditor($content, $local, cls_identifier).refreshPreview();
         };
 
-        DiscussionUtil.processEachMathAndCode = function(text, processor) {
-            var $div, ESCAPED_BACKSLASH, ESCAPED_DOLLAR, RE_DISPLAYMATH, RE_INLINEMATH, cnt, codeArchive, processedText;
-            codeArchive = [];
-            RE_DISPLAYMATH = /^([^\$]*?)\$\$([^\$]*?)\$\$(.*)$/m;
-            RE_INLINEMATH = /^([^\$]*?)\$([^\$]+?)\$(.*)$/m;
-            ESCAPED_DOLLAR = '@@ESCAPED_D@@';
+        var RE_DISPLAYMATH = /^([^\$]*?)\$\$([^\$]*?)\$\$(.*)$/m,
+            RE_INLINEMATH = /^([^\$]*?)\$([^\$]+?)\$(.*)$/m,
+            ESCAPED_DOLLAR = '@@ESCAPED_D@@',
             ESCAPED_BACKSLASH = '@@ESCAPED_B@@';
+
+        /**
+         * Formats math and code chunks
+         * @param safeHtml - post contents in form of safe (escaped and/or stripped) HTML
+         * @param processor - callback to post-process math and code chunks. Should return HtmlUtils.HTML or "subclass"
```

Comment:
See comment above: https://github.com/edx/edx-platform/pull/12914/files#r69921581

I'm not sure this makes sense to return an HtmlUtils.HTML if it doesn't take an HtmlSnippet.  It just seems like the input and output should be consistent, since its purpose doesn't seem to be to convert plain text into HTML.